### PR TITLE
fix: avoid nvidia latency issues by enabling nvidia-persistenced systemd service

### PR DESCRIPTION
**Pull request by alexeldeib**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: parts/linux/cloud-init/artifacts/nvidia-persistenced.service [hunk 1]
```diff
@@ -0,0 +1,13 @@
+[Unit]
+Description=NVIDIA Persistence Daemon
+Wants=syslog.target
+
+[Service]
+Type=forking
```
Comment (marwanad) [inline parts/linux/cloud-init/artifacts/nvidia-persistenced.service:6]: 🔑

File: parts/linux/cloud-init/artifacts/nvidia-persistenced.service [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+ExecStart=/usr/bin/nvidia-persistenced --verbose
+ExecStopPost=/bin/rm -rf /var/run/nvidia-persistenced
+PIDFile=/var/run/nvidia-persistenced/nvidia-persistenced.pid
```
Comment (alexeldeib) [inline parts/linux/cloud-init/artifacts/nvidia-persistenced.service:9]: self-nit: is the pid file usable if it's deleted by ExecStopPost? which happens first?

**Review by ganeshkumar5699 — APPROVED** [review]

**Comment by alexeldeib** [comment]
actually...I should do this inside https://github.com/Azure/aks-gpu/blob/main/install.sh

then it will always apply, and it'll be a little simpler.

File: parts/linux/cloud-init/artifacts/nvidia-persistenced.service (hunk 2 above)
Comment (alexeldeib) [reply parts/linux/cloud-init/artifacts/nvidia-persistenced.service:9]: I got rid of this, it's not in upstream template any more

**Comment by alexeldeib** [comment]
superceded by https://github.com/Azure/AgentBaker/pull/3209

**Pull request closed without merging**